src/Swarm/TUI/Model/StateUpdate.hs

```diff
@@ -98,6 +105,24 @@ scenarioToAppState siPair@(scene, _) userSeed toRun = do
     x' <- liftIO $ a x
     l .= x'
 
+data WhenToPersist
+  = PersistImmediate
+  | PersistDeferred
+
+attainAchievement :: WhenToPersist -> CategorizedAchievement -> EventM Name AppState ()
+attainAchievement persist a = do
+  currentTime <- liftIO getZonedTime
+  (uiState . uiAchievements)
+    %= M.insertWith
+      (<>)
+      a
+      (Attainment a Nothing currentTime)
+  case persist of
+    PersistImmediate -> do
+      newAchievements <- use $ uiState . uiAchievements
+      liftIO $ saveAchievementsInfo $ M.elems newAchievements
+    PersistDeferred -> return ()
```

Comment:
When will deferred achievements actually be saved?  Upon winning a scenario?  I don't think I saw any calls to `saveAchievementsInfo` anywhere else.  What kind of achievements do you envision being deferred vs immediate?

Reply:
I added this speculatively, but I suppose I'll remove it for now.